### ISSUE TITLE
Cursor header can grow very large for /_sql?format=txt endpoint

### ISSUE THREAD

**Issue by mitar** [elastic/elasticsearch#83284]
### Elasticsearch Version

7.16

### Java Version

_bundled_

### OS Version

Docker image

### Problem Description

It seems that cursor header in `_sql?format=txt` endpoint response can grow very large, so large that it hits a limit in `curl`.

### Steps to Reproduce

* Create a mapping with 20k fields for index `test`.
* Run:
```
curl -X GET "localhost:9200/_sql?format=txt" -H 'Content-Type: application/json' -d'
{
  "query": "DESC test"
}
'
```
* Observe `curl` failing with:
```
curl: (27) Rejected 131055 bytes header (max is 102400)!
```

**Comment by Luegg** [comment]
Thanks for reporting the issue. Unfortunately there is no easy work around available at the moment. Because ES SQL itself does not maintain any state, the cursor has to include all state necessary to fetch the next page. In case of `describe <table>` this is the complete result set. While this approach has the downside reported in this issue it ensures that you're getting a consistent view on the columns while scrolling through the data.

An option might be to improve the compression of cursors. This won't solve the issue in general but might give enough leeway to make queries like `describe <table>` more practical with larges indices.

**Comment by Luegg** [comment]
Hey @mitar, this issue is now closed because #83591 should resolve it in most cases (including the one reported by you). The fix will be available with ES 8.2.

**Comment by mitar** [comment]
Awesome. That was a quick resolution. Amazing.
